### PR TITLE
Add: Monthly Dev Post of July 2019

### PR DESCRIPTION
**Pull request by michicc**
To edit or view the current result:
https://github.com/OpenTTD/website/blob/monthly-dev-post/_posts/2019-07-14-monthly-dev-post.md

### CODE REVIEW

**Comment by LordAro** [comment]
file needs a rename :>

**Comment by michicc** [comment]
Copy editing and comments sorely needed.

File: _posts/2019-06-01-monthly-dev-post.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+If you want to modify the behaviour of trains, looking into the files whose names start with train is certainly not a bad idea.
+Of course, such a strict division is impossible to keep in all cases.
+The code that decides what should happen if an aircraft stops at a station conceptually belongs to both `aircraft` and `station`, even if it can exist in only one place in reality.
+
+You might have also noticed that the file names follow a pattern and that you can find similarly named files in each group.
+Many files are named after the following schema: `<group>_<suffix>.<extension>`.
+The `<group>` part was already explained above.
+The `<suffix>` tells you something about the contents of the file without having to open it inside an editor and can be viewed as a kind of *file type*.
+The `<extension>` part is for the benefit of the C++ compiler and denotes if the file is a source code file that is passed to the compiler (`.cpp`) or a header file (`.h`) that is included in other source code files.
+There's also the hybrid of a header file containing source code (marked `.hpp`), which is has to be this way due to how C++ templates work.
```
Comment (frosch123) [inline _posts/2019-06-01-monthly-dev-post.md]: "which is has ..." -> "for C++ template implementations"

File: _posts/2019-07-14-monthly-dev-post.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+Every so often, "I want to make this thing but I don't know how" comes up.
+
+And indeed, looking at the [source code](https://github.com/OpenTTD/OpenTTD.git) of OpenTTD for the first time can be a daunting task.
+The `src` folder alone contains 1314 files, and is accompanied by a plenthora of other ancilliary files.
+To make it less daunting to get started with realising your idea, we will explore various parts of the OpenTTD source code and try to build some kind of index to the code.
+We are going to assume that you have seen source code before and have some familiarity with the C++ language.
+There are thousands of introductory tutorials on programming in general and specific to C++ on the internet, thus we don't think it would make sense to replicate them all here.
+Also, If you've never compiled OpenTTD before, we've got [general tips](https://github.com/OpenTTD/OpenTTD/blob/master/README.md#70-compiling) and some specific to [Windows](https://github.com/OpenTTD/OpenTTD/blob/master/docs/Readme_Windows_MSVC.md).
+
+This is planed to be a multi-part series, to be continued as time permits.
```
Comment (nikolas) [inline _posts/2019-07-14-monthly-dev-post.md]: planed -> planned

File: _posts/2019-07-14-monthly-dev-post.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+Inspiration for this comes from an interesting treaty by a fellow called MaiZure: [Decoded: OpenTTD](http://www.maizure.org/projects/decoded-openttd/index.html).
+It's a nice read if you are interested in the technical innards of the game and already somewhat familiar with the source code.
+Unfortunately, if you are not, the parts MaiZure focuses on aren't the parts you'll predominantely need to be able to write your first working code patch.
```
Comment (nikolas) [inline _posts/2019-07-14-monthly-dev-post.md]: predominantely -> predominantly

**Pull request merged**